device.c

```diff
@@ -39,10 +39,30 @@
 #include "stream_out.h"
 #include "stream_in.h"
 
+/* Number of supported input devices. Should be in sync with
+ * number of input devices in audio_policy_configuration.xml */
+#define NUMBER_OF_DEVICES_IN 3
+
 /* Number of supported output devices. Should be in sync with
  * number of output buses in audio_policy_configuration.xml */
 #define NUMBER_OF_DEVICES_OUT 8
 
+
+typedef struct xa_device_map
+{
+    audio_devices_t device_type;  /* something like AUDIO_DEVICE_OUT_BUS */
```

Comment:
I would put 'one of the AUDIO_DEVICE_OUT_XXX'

Reply:
Done.